components/telemetry-operator/internal/kubernetes/utils_test.go

```diff
@@ -162,3 +164,61 @@ func TestMergeChecksumAnnotations(t *testing.T) {
 	require.Equal(t, desired.Annotations["checksum/3"], "3")
 	require.Equal(t, desired.Annotations["checksum/4"], "4")
 }
+
+func TestCreateOrUpdateClusterRoleBinding(t *testing.T) {
+	mockClient := &mocks.Client{}
+	notFoundErr := errors.NewNotFound(schema.GroupResource{}, "")
+	mockClient.On("Get", mock.Anything, mock.Anything, mock.Anything).Return(notFoundErr)
+	mockClient.On("Create", mock.Anything, mock.Anything).Return(nil)
+
+	resClusterRoleBinding := rbac.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo",
+			Namespace: "foo-namespace",
+		},
+		Subjects: []rbac.Subject{{Name: "telemetry-fluent-bit", Namespace: "kyma-system", Kind: "ServiceAccount"}},
+		RoleRef: rbac.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     "telemetry-fluent-bit",
+		},
+	}
+	err := CreateOrUpdateClusterRoleBinding(context.Background(), mockClient, &resClusterRoleBinding)
+
+	require.NoError(t, err)
+}
+
+func TestCreateOrUpdateClusterRole(t *testing.T) {
+	mockClient := &mocks.Client{}
+	notFoundErr := errors.NewNotFound(schema.GroupResource{}, "")
+	mockClient.On("Get", mock.Anything, mock.Anything, mock.Anything).Return(notFoundErr)
+	mockClient.On("Create", mock.Anything, mock.Anything).Return(nil)
+
+	clusterRole := rbac.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo",
+			Namespace: "foo-namespace",
+		},
+		Rules: []rbac.PolicyRule{{Verbs: []string{"get", "list", "watch"}, APIGroups: []string{""}, Resources: []string{"namespaces", "pods"}}},
+	}
+	err := CreateOrUpdateClusterRole(context.Background(), mockClient, &clusterRole)
+
+	require.NoError(t, err)
+}
+
+func TestCreateOrUpdateServiceAccount(t *testing.T) {
+	mockClient := &mocks.Client{}
+	notFoundErr := errors.NewNotFound(schema.GroupResource{}, "")
+	mockClient.On("Get", mock.Anything, mock.Anything, mock.Anything).Return(notFoundErr)
+	mockClient.On("Create", mock.Anything, mock.Anything).Return(nil)
+
+	resServiceAcc := corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo",
+			Namespace: "foo-namespace",
+		},
+	}
+	err := CreateOrUpdateServiceAccount(context.Background(), mockClient, &resServiceAcc)
```

Comment:
Just wondering - what are we testing here?

Reply:
Yeah true everything is mocked will remove this

Reply:
Switching to the fake client from controller runtime might make this test a bit more meaningful.

Reply:
No-no, it might make sense to test that either `Create` or `Update` is called on the mock.

Reply:
@chrkl Fully agree